subsys/mgmt/updatehub/updatehub_integrity.c

```diff
@@ -18,43 +24,22 @@ int updatehub_integrity_init(struct updatehub_crypto_context *ctx)
 		return -EINVAL;
 	}
 
-	memset(ctx, 0, sizeof(struct updatehub_crypto_context));
-
-#if defined(CONFIG_FLASH_AREA_CHECK_INTEGRITY_MBEDTLS)
-	ctx->md_info = mbedtls_md_info_from_type(MBEDTLS_MD_SHA256);
-	if (ctx->md_info == NULL) {
-		LOG_DBG("Message Digest not found or not enabled");
-		return -ENOENT;
-	}
-
-	mbedtls_md_init(&ctx->md_ctx);
-	ret = mbedtls_md_setup(&ctx->md_ctx, ctx->md_info, 0);
-	if (ret == MBEDTLS_ERR_MD_BAD_INPUT_DATA) {
-		LOG_DBG("Bad Message Digest selected");
-		return -EFAULT;
-	}
-	if (ret == MBEDTLS_ERR_MD_ALLOC_FAILED) {
-		LOG_DBG("Failed to allocate memory");
-		return -ENOMEM;
-	}
-
-	ret = mbedtls_md_starts(&ctx->md_ctx);
-	if (ret == MBEDTLS_ERR_MD_BAD_INPUT_DATA) {
-		LOG_DBG("Bad Message Digest selected");
-		return -EFAULT;
-	}
-#elif defined(CONFIG_FLASH_AREA_CHECK_INTEGRITY_TC)
-	ret = tc_sha256_init(&ctx->sha256sum);
-	if (ret != TC_CRYPTO_SUCCESS) {
-		LOG_DBG("Invalid integrity context");
+#if defined(CONFIG_MBEDTLS_PSA_CRYPTO_CLIENT)
```

Comment:
I think that 
```suggestion
#if defined(CONFIG_PSA_CRYPTO_CLIENT)
```
would be better as you already did in `subsys/mgmt/updatehub/Kconfig`.
The same holds for other occurrences of the `CONFIG_MBEDTLS_PSA_CRYPTO_CLIENT`.

Reply:
True. That was an oversight; well spotted. I'll replace the occurrences in this file.